pkg/asset/installconfig/ibmcloud/metadata.go

```diff
@@ -85,14 +91,52 @@ func (m *Metadata) SetCISInstanceCRN(crn string) {
 	m.cisInstanceCRN = crn
 }
 
+// ComputeSubnets gets the Subnet details for compute subnets
+func (m *Metadata) ComputeSubnets(ctx context.Context) (map[string]Subnet, error) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	if len(m.ComputeSubnetNames) > 0 && len(m.computeSubnets) == 0 {
+		client, err := m.Client()
+		if err != nil {
+			return nil, err
+		}
+		m.computeSubnets, err = getSubnets(ctx, client, m.Region, m.ComputeSubnetNames)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return m.computeSubnets, nil
+}
+
+// ControlPlaneSubnets gets the Subnet details for control plane subnets
+func (m *Metadata) ControlPlaneSubnets(ctx context.Context) (map[string]Subnet, error) {
+	m.mutex.Lock()
+	defer m.mutex.Unlock()
+
+	if len(m.ControlPlaneSubnetNames) > 0 && len(m.controlPlaneSubnets) == 0 {
+		client, err := m.Client()
+		if err != nil {
+			return nil, err
+		}
+		m.controlPlaneSubnets, err = getSubnets(ctx, client, m.Region, m.ControlPlaneSubnetNames)
+		if err != nil {
+			return nil, err
+		}
+	}
+
+	return m.controlPlaneSubnets, nil
+}
+
 // Client returns a client used for making API calls to IBM Cloud services.
 func (m *Metadata) Client() (*Client, error) {
 	if m.client == nil {
 		client, err := NewClient()
-		client.SetVPCServiceURLForRegion(context.TODO(), m.Region)
 		if err != nil {
 			return nil, err
 		}
+		client.SetVPCServiceURLForRegion(context.TODO(), m.Region)
```

Comment:
does access to client need to be wrapped in a mutex here as done in ControlPlaneSubnets and ComputeSubnets?

Reply:
That's not a bad idea, I can add that.

Reply:
added